explorer/src/Pos/Explorer/Web/Server.hs

```diff
@@ -186,6 +193,20 @@ explorerHandlers _sendActions =
 -- API Functions
 ----------------------------------------------------------------
 
+getTotalAda :: ExplorerMode ctx m => m Micro
+getTotalAda = do
+    utxoSum <- EX.getUtxoSum
+    validateUtxoSum utxoSum
+    pure $ fromInteger utxoSum / 1e6
+  where
+    validateUtxoSum :: ExplorerMode ctx m => Integer -> m ()
+    validateUtxoSum n
+        | n < 0 = throwM $ Internal $
+            sformat ("Internal tracker of utxo sum has a negative value: "%build) n
+        | n > coinToInteger (maxBound :: Coin) = throwM $ Internal $
```

Comment:
@gromakovsky we are sure that utx sum will always fit into `Coin` , right ?

Because it seems to me that it is safer to throw exception here then to bound it.

@Pastafarianist said
> When I implemented it, I was thinking that since this was going to be displayed on CoinMarketCap, we probably wanted to make that endpoint as robust as possible — thus never return errors, but sometimes return invalid values instead.

but, as in practice, it should fit `Coin` in this corner use case (which I don't think it will happen) I would rather fail then show incorrect information about total ada.

What do you think @gromakovsky ?

Reply:
I believe that there is a function which checks that coin is in bounds, smth like `integerToCoin`.
Please use it

Reply:
> we are sure that utx sum will always fit into Coin , right ?

We are sure only if we are sure that we have no bugs. So we are not sure.

> I would rather fail then show incorrect information about total ada.

But that's exactly what is done here, isn't it? `throwM Internal` looks quite reasonable.

Reply:
> But that's exactly what is done here, isn't it? throwM Internal looks quite reasonable.

right - don't know how I missed it. I thought it returns truncated Coin. Its ok. Sorry @Pastafarianist I missread the line